$NetBSD: patch-configure.ac,v 1.7 2024/03/24 12:46:23 adam Exp $

install .oct loadable modules with INSTALL_LIB to avoid stripping them

Allow overriding MKOCTFILE_F77 so we can install a full path gfortran
(for building octave-forge packages like control)

Portability fixes.

--- configure.ac.orig	2024-03-12 18:00:23.000000000 +0000
+++ configure.ac
@@ -238,8 +238,11 @@ AC_PROG_LN_S
 AC_PROG_MKDIR_P
 
 AC_PROG_INSTALL
-INSTALL_SCRIPT="${INSTALL}"
+test -z "$INSTALL_SCRIPT" && INSTALL_SCRIPT='${INSTALL}'
 AC_SUBST(INSTALL_SCRIPT)
+test -z "$INSTALL_LIB" && INSTALL_LIB='${INSTALL_SCRIPT}'
+AC_SUBST(INSTALL_LIB)
+
 
 ## Programs used when running Octave.
 OCTAVE_PROG_GHOSTSCRIPT
@@ -933,11 +936,11 @@ if test "$cross_compiling" = yes && test
   MKOCTFILE_F77=`echo "$F77" | $SED "s,$CROSS_TOOL_PREFIX,,"`
   MKOCTFILE_RANLIB=`echo "$RANLIB" | $SED "s,$CROSS_TOOL_PREFIX,,"`
 else
-  MKOCTFILE_AR="$AR"
-  MKOCTFILE_CC="$CC"
-  MKOCTFILE_CXX="$CXX"
-  MKOCTFILE_F77="$F77"
-  MKOCTFILE_RANLIB="$RANLIB"
+  MKOCTFILE_AR="${MKOCTFILE_AR-$AR}"
+  MKOCTFILE_CC="${MKOCTFILE_CC-$CC}"
+  MKOCTFILE_CXX="${MKOCTFILE_CXX-$CXX}"
+  MKOCTFILE_F77="${MKOCTFILE_F77-$F77}"
+  MKOCTFILE_RANLIB="${MKOCTFILE_RANLIB-$RANLIB}"
 fi
 AC_MSG_NOTICE([defining CROSS_TOOL_PREFIX to be $CROSS_TOOL_PREFIX])
 AC_MSG_NOTICE([defining MKOCTFILE_AR to be $MKOCTFILE_AR])
@@ -2365,7 +2368,7 @@ fi
 
 ### Check for SUNDIALS library features, some required, some optional.
 
-if test "x$octave_have_sundials_libs" == xyes; then
+if test "x$octave_have_sundials_libs" = xyes; then
 
   CPPFLAGS="$SUNDIALS_IDA_CPPFLAGS $SUNDIALS_NVECSERIAL_CPPFLAGS $KLU_CPPFLAGS $BLAS_CPPFLAGS $CPPFLAGS"
   LDFLAGS="$SUNDIALS_IDA_LDFLAGS $SUNDIALS_NVECSERIAL_LDFLAGS $KLU_LDFLAGS $BLAS_LDFLAGS $LDFLAGS"
@@ -2405,7 +2408,7 @@ dnl FIXME: The test on the sunlinsol_klu
 dnl How can we do a better job here?  Do we need to disable sundials
 dnl any tests fail, or can we fix __ode15__.cc so that it still partially
 dnl works when some things are missing (for example, KLU)?
-if test "x$octave_have_sundials_libs" == xyes; then
+if test "x$octave_have_sundials_libs" = xyes; then
   if test "x$octave_cv_sundials_sunlinsol_dense" = xyes \
       && test "x$octave_cv_sundials_realtype_is_double" = xyes \
       && test "x$octave_have_sundials_compatible_api" = xyes; then
